### PR TITLE
Notifications telemetry + hardening

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorageManager.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    }
+  }
+
+  private static Gson createGson() {
+    return new GsonBuilder()
+      .registerTypeAdapter(OffsetDateTime.class, new OffsetDateTimeAdapter())
+      .create();
+  }
+
+  private TelemetryLocalStorage load() throws IOException {
```
Comment (damien-urruty-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorageManager.java:106]: I am not sure I understand the difference between read and load. When to use one or the other ? Do we need both ? If so maybe we could come up with better names

**Review by damien-urruty-sonarsource — APPROVED** [review]
Looks good to me, it simplifies with the need to merge that is removed, so kudos :+1: 

We still have a distinction between data that is accumulated and instant data, but I think it's ok.

I made a few comments, and I still have a doubt about the names of some concepts, but I can't suggest better names

File: core/src/main/java/org/sonarsource/sonarlint/core/container/model/DefaultSonarQubeNotification.java [hunk 2]
```diff
@@ -23,10 +23,10 @@
 
 import javax.annotation.concurrent.Immutable;
 
-import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotification;
+import org.sonarsource.sonarlint.core.client.api.notifications.ServerNotification;
 
 @Immutable
-public class DefaultSonarQubeNotification implements SonarQubeNotification {
+public class DefaultSonarQubeNotification implements ServerNotification {
```
[the remark this answers is not in the document]
Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/container/model/DefaultSonarQubeNotification.java:29]: Good catch !

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorageManager.java (hunk 1 above)
Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorageManager.java:106]: One method (`read`) is reading from the same FileChannel used later by write (in order to ensure atomicity). 
`load` is just reading without lock (mainly used to check the isEnabled flag).

Let me try to rename to make it more explicit.

**Review by damien-urruty-sonarsource — APPROVED** [review]

**Pull request merged**